pkg/vk8s/helpers.go

```diff
@@ -28,19 +28,31 @@ import (
 )
 
 // getPasswordFromSecret retrieves the password from the secret using the provided key
-func getPasswordFromSecret(secret map[string][]byte, key string) (string, error) {
+func getPasswordFromSecret(secret map[string][]byte, key string) (*string, error) {
+	var passwd string
 	pwd, ok := secret[key]
+	passwd = string(pwd)
 	if !ok {
-		return "", fmt.Errorf("password not found, secret must have a key with name %q", key)
+		return nil, fmt.Errorf("password not found, secret must have a key with name %q", key)
 	}
-	return string(pwd), nil
+	return &passwd, nil
 }
 
 // GetSuperuserPassword returns the superuser password if it has been provided
 func GetSuperuserPassword(ctx context.Context, cl client.Client, log logr.Logger,
-	e events.EVWriter, vdb *vapi.VerticaDB) (string, error) {
-	if vdb.Spec.PasswordSecret == "" {
-		return "", nil
+	e events.EVWriter, vdb *vapi.VerticaDB) (*string, error) {
+	return GetSuperuserPasswordForUpdate(ctx, cl, log, e, vdb, false)
```

Comment:
It would have been easier to make the secret a parameter. The function should get a password secret name and get the secret. Any other check must be done by the caller.